crypto/zk/artifacts/artifacts.go

```diff
@@ -227,6 +229,16 @@ func downloadFile(ctx context.Context, fromUrl, toPath string) error {
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("error on download file %s: http status: %d", fromUrl, resp.StatusCode)
+	}
+
+	out, err := os.Create(toPath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
```

Comment:
nit: you should check this file error close by leaving the defer here and adding this at the end of the func:

```
if err := out.Close(); err != nil {
    return err
}
```

This works because closing a file is idempotent: you can do it twice and it's okay.

Reply:
why this matters: if there's an error when flushing writes to disk, Close might error, so you don't want to ignore that.